x/ccv/provider/keeper/grpc_query.go

```diff
@@ -266,11 +266,13 @@ func (k Keeper) QueryConsumerChainsValidatorHasToValidate(goCtx context.Context,
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// get all the consumer chains for which the validator is opted-in
+	// get all the consumer chains for which the validator is already or will be
+	// opted-in starting from the next epoch
 	consumersToValidate := []string{}
 	for _, consumer := range k.GetAllConsumerChains(ctx) {
 		chainID := consumer.ChainId
-		if k.IsConsumerValidator(ctx, chainID, types.NewProviderConsAddress(consAddr)) || k.IsOptedIn(ctx, chainID, types.NewProviderConsAddress(consAddr)) {
+
+		if k.IsOptedIn(ctx, chainID, types.NewProviderConsAddress(consAddr)) {
```

Comment:
Note that top N validators are only opted in right when the epoch begins, so this doesn't tell you who will get opted into the chain because of being in the top N.

Of course we cannot know for sure, but I would change this to just compute the MinPowerToOptIn for the chain, and then compute the validators that would be opted in, if the epoch would end without further power changes

Reply:
Good point. Will add that. Thanks!